pallets/api/src/fungibles/tests.rs

```diff
@@ -675,3 +675,95 @@ mod read_weights {
 		assert_eq!(token_exists.proof_size(), 3675);
 	}
 }
+
+mod ensure_codec_indexes {
+	use super::{Encode, RuntimeCall, *};
+	use crate::{fungibles::Call::*, mock::RuntimeCall::Fungibles};
+	#[test]
+	fn ensure_read_variant_indexes() {
+		let total_supply = TotalSupply::<Test>(Default::default());
```

Comment:
Could we do something like to prevent the repetition? Makes it easier to add/maintain.

```
[
  (TotalSupply::<Test>(Default::default()).encode(), "TokenSupply"),
  (TokenName::<Test>(Default::default()), "TokenName"),
].all(|(encoded_variant, name)| { 
  assert_eq!(encoded_variant[0], 0, "{name} variant index changed");
})
```

Reply:
nice. See latest commit.